File: Miscellaneous_SoleKeyEncryption.cpp
#include "stdafx.h"
#include "SoleKeyEncryption.h"

#include <iostream>

using namespace std;

int main()
{
	char message[] = "merry xmas";
	unsigned i = 0;
	vector<char> msg = {};
	while (message[i] != '\0') {
		msg.push_back(message[i++]);
	}
	msg.shrink_to_fit();

	char key[] = "A5FB17C4D8"; // you might want to avoid zeroes
	i = 0;
	vector<char> k = {};
	while (key[i] != '\0') {
		k.push_back(key[i++]);
	}
	k.shrink_to_fit();

	SoleKeyEncryption go_secure;

	vector<string> encrypted = go_secure.encodeWord(msg, k);
	string disp = "";
	for (string s : encrypted) {
		disp += s + "; ";
	}
	cout << "Message is '" << string(message) << "';\nEncrypted version is " << disp << endl;

	string decrypted = go_secure.decodeWord(encrypted, k);
	cout << "\n\nDecrypted version is '" << decrypted << "'.";

	return 0;
}

File: Miscellaneous_SoleKeyEncryption.java
package miscellaneous;

import java.util.Arrays;

public class Miscellaneous {

    public static void main(String[] args) {
        char[] message = "merry xmas".toCharArray();
        char[] key = "A5FB17C4D8".toCharArray(); // you might want to avoid zeroes
        SoleKeyEncryption go_secure = new SoleKeyEncryption();
        
        String[] encrypted = go_secure.encodeWord(message, key);
        System.out.println("Message is '" + String.valueOf(message) +
                "';\nEncrypted version is " + Arrays.toString(encrypted));
        
        String decrypted = go_secure.decodeWord(encrypted, key);
        System.out.println("\n\nDecrypted version is '" + decrypted + "'.");
    }

}